modules/ConnectionManager.py

```diff
@@ -0,0 +1,212 @@
+                logger.exception("Exception in Poller Thread: %s", e)
+
+
+class ConnectionManager(object):
+    def __init__(self, creds, config):
+        self.creds = creds
+        self.handles = {}
+        self.config = config
+        self._collectors = []
+        self._poll_threads = {}
+        # temporary blacklist of unreachable servers
+        self.blacklist = {}
+
+    def get_inventory(self):
+        """
+        Get updated inventory
+        :return: list of servers
+        """
+        config = self.get_config_data()
+        server_list = []
+        if 'netbox' in config and len(config['netbox']):
+            from modules.Netbox import Netbox
+
+            netbox_data = config['netbox']
+            netbox_obj = Netbox(nb_config=netbox_data)
+            server_list += netbox_obj.get_ucsm_servers_from_regions(netbox_data["regions"])
+        if 'servers' in config and type(config['servers']) == list:
+            server_list += config['servers']
+        logger.debug("Serverlist: %s" %server_list)
+        return server_list
+
+    def get_handle(self, host):
+        """
+        Returns the handle for host
+        """
+        return self.handles.get(host, None)
+
+    def get_handles(self):
+        return self.handles
+
+    def update_handle(self, server):
+        """
+        Checks if the handle is correct for host. Returns true on success.
+        """
+        rv = False
+        # check if server is currently unreachable
+        if server in self.blacklist:
+            if self.blacklist[server] <= time.time():
+                del self.blacklist[server]
+            else:
+                return False
+
+        if server in self.handles:
+            try:
+                self.handles[server].query_dn("sys")
+                logger.debug("test query to %s successful" % server)
+                rv = True
+            except UcsException as e:
+                if e.error_code == 552:
+                    logger.debug("Refresh login to %s" % server)
+                else:
+                    logger.error("Problem refreshing %s:%s",
+                                server, str(e), extra={'server': server, 'error': e})
+                try:
+                    rv = False
+                except: pass
+
+        if server not in self.handles:
+            logger.debug("Login into %s" % server)
+            srv_obj = UcsmServer(server, self.creds['username'], self.creds['master_password'], self.config['domain'])
+            if not srv_obj.handle:
+                if self.config['retry_timeout'] > 0:
+                    self.blacklist[server] = time.time() + self.config['retry_timeout']
+                    logger.error("Problem creating login to %s. Retry in %s seconds",
+                                server, self.config['retry_timeout'], extra={'server': server})
+                else:
+                    logger.error("Problem creating login to %s", server, extra={'server': server})
```

Comment:
Don't we want to return in both cases? I will always run into `handle.login` exception. I mean it is catched, but can be avoided.

Reply:
Returning value.